menpo/shape/pointcloud.py

```diff
@@ -777,49 +780,186 @@ def _view_3d(self, figure_id=None, new_figure=False):
             The id of the figure to be used.
         new_figure : `bool`, optional
             If ``True``, a new figure is created.
+        render_markers : `bool`, optional
+            If ``True``, the markers will be rendered.
+        marker_style : `str`, optional
+            The style of the markers.
+            Example options ::
+
+                {2darrow, 2dcircle, 2dcross, 2ddash, 2ddiamond, 2dhooked_arrow,
+                 2dsquare, 2dthick_arrow, 2dthick_cross, 2dtriangle, 2dvertex,
+                 arrow, axes, cone, cube, cylinder, point, sphere}
+
+        marker_size : `float` or ``None``, optional
+            The size of the markers. This size can be seen as a scale factor
+            applied to the size markers, which is by default calculated from
+            the inter-marker spacing. If ``None``, then an optimal marker size
+            value will be set automatically.
+        marker_colour : See Below, optional
+            The colour of the markers.
+            Example options ::
+
+                {r, g, b, c, m, k, w}
+                or
+                (3, ) ndarray
+
+        marker_resolution : `int`, optional
+            The resolution of the markers. For spheres, for instance, this is
+            the number of divisions along theta and phi.
+        step : `int` or ``None``, optional
+            If `int`, then one every `step` vertexes will be rendered.
+            If ``None``, then all vertexes will be rendered.
+        alpha : `float`, optional
+            Defines the transparency (opacity) of the object.
+        render_numbering : `bool`, optional
+            If ``True``, the points will be numbered.
+        numbers_colour : See Below, optional
+            The colour of the numbers.
+            Example options ::
+
+                {r, g, b, c, m, k, w}
+                or
+                (3, ) ndarray
+
+        numbers_size : `float` or ``None``, optional
+            The size of the numbers. This size can be seen as a scale factor
+            applied to the numbers, which is by default calculated from
+            the inter-marker spacing. If ``None``, then an optimal numbers size
+            value will be set automatically.
 
         Returns
         -------
-        viewer : PointCloudViewer3d
-            The Menpo3D viewer object.
+        renderer : `menpo3d.visualize.PointGraphViewer3d`
+            The Menpo3D rendering object.
         """
         try:
-            from menpo3d.visualize import PointCloudViewer3d
-            return PointCloudViewer3d(figure_id, new_figure,
-                                      self.points).render()
+            from menpo3d.visualize import PointGraphViewer3d
+            edges = np.empty(0)
+            renderer = PointGraphViewer3d(figure_id, new_figure,
+                                          self.points, edges)
+            renderer.render(
+                render_lines=False, line_colour='r', line_width=4,
```

Comment:
Are the line width and color supposed to be hardcoded here? Surely they aren't even used? Shouldn't we just ignore these arguments?